components/messenger/transports.rst

```diff
@@ -0,0 +1,225 @@
+.. index::
+    single: Messenger; Transports
+
+Transports
+==========
+
+To send and receive messages via message brokers, the Messenger component has
+"transports". Transports are responsible for routing messages to and from
+the message brokers.
+
+Every transport is configurable using a DSN. This DSN allows you to chose the
+transport layer as well as configure it.
+
+AMQP
+----
+
+The most famous message broker protocol is probably AMQP, which is most
+commonly implemented with RabbitMQ. The Messenger component has built-in
+support for AMQP.
+
+How it works?
+~~~~~~~~~~~~~
+
+A DSN protocol that uses ``amqp`` is recognized and sent to the built-in AMQP transport.
+For example::
+
+    $dsn = 'amqp://user:password@localhost/%2f/messages'
+
+The messages will be sent to the ``messages`` exchange bound to the ``messages``
+queue on the ``/`` vhost (the ``%2f`` is a url-encoded ``/``).
+
+.. note:
+
+    By default, RabbitMQ uses ``guest`` as a username and ``guest`` as a password
+    and has a ``/`` vhost.
+
+Error handling
+~~~~~~~~~~~~~~
+
+If something wrong happens (i.e. an exception is thrown) while handling your message,
+the default behaviour is that your message will be "NACK" and requeued.
+
+However, if your exception implements the ``RejectMessageExceptionInterface`` interface,
+the message will be rejected from the queue.
+
+Retry
+~~~~~
+
+When receiving a message from a broker, it might happen that some exceptions will
+arise. Typically, a 3rd party provider is down or your system is under heavy load
+and can't really process some messages. To handle this scenario, there is a built-in
+retry mechanism that can be enabled via your DSN::
+
+    amqp://guest:guest@localhost/%2f/messages
```

Comment:
this looks like an `sprintf` but no value given for `%2f` 🤔 

Please use `sprintf` with a value or use `.. code-block:: terminal` instead

Reply:
This is the correct string - no sprintf - it’s an Amqp thing - the vhost I believe

Reply:
Ok but then no :: php code block

.. code-block:: bash ???